src/common/summarize-profile.js

```diff
@@ -108,23 +94,51 @@ export function summarizeProfile(profile: Profile) {
  * are cached between calls.
  * @returns {function} Function categorizer.
  */
-function functionNameCategorizer() {
-  const cache = new Map();
-  return function functionNameToCategory(name) {
-    const existingCategory = cache.get(name);
-    if (existingCategory !== undefined) {
+function functionNameCategorizer(numFunctions): (number, number[], string[]) => string {
+  const cache = new Array(numFunctions);
+  cache.fill('', 0, numFunctions);
```

Comment:
also I don't think this is useful to precise the length here, as we don't use the length anywhere; just use `const cache = []`.